turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1199,6 +1250,13 @@ pub(crate) async fn analyse_ecmascript_module_internal(
                 span: _,
                 in_try: _,
             } => {
+                if done
+                    .iter()
+                    .any(|done_ast_path| ast_path.starts_with(done_ast_path))
```

Comment:
That looks like you can easily run in a bad case for performance since performance scales N^2 with the number of `R.pipe` expressions in the module.

Reply:
You can nest effects similar to `Effect::Conditional` does it. So `Effect::Member` would have an `effects` prop that is only added to the queue if `Effect::Member` is not directly applied.